Fixes for 1.0.14 SDK release (v2)

zangold opened the pull request:
- merge v1.3.268
- fix bugs that the above merge caused
- allow json_gen layer to be built independently of api_dump

charles-lunarg committed:
apidump: Use setfill/setw to prevent indentation overflow

When the indentation gets too high apidump would start reading other memory in the executable,
which leads to humerous but highly inccorect output.

Change-Id: I94e31fc994bb373a29804c9c55a257cb6193266b

christophe-lunarg committed:
vkconfig: fix application controlled mode

Change-Id: I66b15282db99045a7c5105e2d9f7a977bf0bdb41

andrew-lunarg committed:
Removed DevSim references.

TonyBarbour committed:
build: Update to 1.3.234 headers

lunarpapillo committed:
scripts: use python3 if python is not available

Some modern Linux distributions come with the "python3" executable but
not the "python" executable.  This change allows whichever is present
on the invoking machine to be used.

lunarpapillo committed:
vkconfig: include <limits>

vkconfig_core/ uses std::numeric_limits<T>, but doesn't explicitly
include <limits>, which defines this.

This seemed to work fine on older Linux distributions (e.g. Ubuntu 16.04),
probably because the header file was implicitly included somewhere;
but it fails on newer distributions (e.g. Ubuntu 22.04).

Always explicitly including <limits> is painless and lets the project
build on all Linux versions.

TonyBarbour committed:
build: Update to 1.3.235 headers

Also disable VK_EXT_descriptor_buffer because VVL did and we use that
dispatch table

charles-lunarg committed:
apidump: Remove variadic templates and inline

Many functions had inline which did nothing. Best to remove it.

Variadic templates were used to pass information into printers that didn't have
the requisite information available to correctly print (or skip) fields. The
solution of variadic templates is unwieldly because every function needed the
template arguments even if they didn't use it.

Thus it is easier to create bespoke variables in the ApiDumpInstance to store
the data before calling the printing functions, then retrieve the data from the
instance. This solution allows simplifying the pNext printing code to no longer
need workarounds for functions which used the variadic arguments.

charles-lunarg committed:
apidump: Factor out head & body print functions

api_dump.cpp included a lot of redundant functions to print the head
and body of the API functions. By creating API function agnostic head
printer and moving the logic of the body printer into the parent, many
lines of code were removed from the generated file.

charles-lunarg committed:
apidump: Remove unsued std::ostream& return type

Returning the stream did not serve any purpose as the stream could be easily
accessed from the ApiDumpSettings which gets passed into every printing
function. This cleans up a lot of the code to be a lot clearer since it is a
form of dead code elimination.

charles-lunarg committed:
apidump: remove dump_bitmaskOption & cleanup OutputAddress

dump_bitmaskOption's logic can be easily done in line, removing a ton
of potential function calls.

Whether OutputAddress should put quotes is purely determined by whether
it is being called within the JSON output.

charles-lunarg committed:
apidump: Refactor generator to use dictionaries

This greatly speeds up the time to generate because we can easily check if any
thing is in a set without iterating the entire list. Rough estimate of build
time improvements is 115 seconds down to 35, or by around a factor of 3x.

christophe-lunarg committed:
vkconfig: add warn_on_robust_oob val setting

Change-Id: I9f4ff61c161867879d8c2676bd36634f4f0461eb

christophe-lunarg committed:
C.I.: fix Github actions build

Change-Id: I1c3b7d7843f1f7e8531890828c1e10778e85315d

christophe-lunarg committed:
vkconfig: improve handling of robust oob warn

Change-Id: Ifd74cd529d9ab289b878381bcdf55aca6fb05637

christophe-lunarg committed:
vkconfig: fix setting dependences with enums

Change-Id: If2fe7d3e71afc7873859e118624aabf83615778a

charles-lunarg committed:
apidump: Use correct name of types

Previously, the logic assumed that the typename did not contain the name
of a member or parameter ofa struct, which is an assumption which doesn't
hold across the API. This resulted in the output printing incorrect names
for many things where the name of the member/parameter is a substring
of the type. For example, 'xcb_connection_t' is the type while 'connection'
is the name. API Dump would print the type for 'connection' as 'xcb_'.

charles-lunarg committed:
apidump: Remove runtime json comma printing

The origina JSON output used a runtime boolean to keep track of whether
to print commas between members & parameters in structs and functions.
This was done because there was no easy way to print something conditional
on the current index. By making the index a part of the member &
parameter & union elements, these runtime bools can be done away with.

charles-lunarg committed:
apidump: Correctly quote json in unknown pnext struct

When printing pNext chains, correctly print the quotes necessary for
the json output to be valid.

TonyBarbour committed:
build: Build layers using C++ 17

Dependencies from Vulkan-ValidationLayers now contain C++ 17

TonyBarbour committed:
docs: Remove VS 2015 references

charles-lunarg committed:
apidump: Fix whitespace between names & types

setw() will reset back to 0 in many cases, one of which includes <<
This meant that call to `stream << setw();` does not apply the proper
whitespace required, leading to not adding any space between the name
and type. `setw() << "";` forces the correct width to be printed.

lunarpapillo committed:
android: limit parallel builds

The Android build is quite aggressive, attempting to consume all available
cores; large builds (particularly the api_dump layer) cause all the processes
involved to consume 3+G of memory, causing physical and swap memory to be
exhausted, which causes sporadic build failures as the OOM killer kills
the compiler processes.

These changes allow the use of an environment variable $VT_BUILD_CORES
to limit how many cores are in use; it also prevents the build from using
additional parallel jobs if the load average is too high (on one 8-core
system, the load average climbed to over 26 before the OOM killer started
killing processes).

charles-lunarg committed:
apidump: Use const correctly in pNext printing

The pNext chain printer for the text output didn't forward the fact that
structs sometimes have void* and other times have const void* for their
pNext chain. Forward the information allows for more accurate printing.

Also fix issue where vkQueuePresentKHR had a return type 'vVkResult'.

charles-lunarg committed:
apidump: Remove unnecessary struct declarations

The only thing 'needing' function declarations is the pNext printer, because vk.xml
is designed to have all the type dependencies encoded in the order the types appear
in the xml. However, since api dump prints unions separately, the unions at least
need to be forward declared for the struct printers. By forward declaring the pNext
printing functions, it is possible to remove all of the struct forward declarations.

charles-lunarg committed:
apidump: Indent struct generator template

Because there are several levels of nested if statements, it became
very difficult to read the templating code. Thus, indenting the if
statements to their corresponding level visually clarifies the already
complex generation logic.

charles-lunarg committed:
apidump: Better handle array member special cases

VkPhysicalDeviceMemoryProperties and VkPhysicalDeviceGroupProperties contain
fixed length arrays whose actual length is determined by another struct member.
Because the vk.xml does not contain the required information, we had to special
case the generation for these structs. Now, instead of hardcoding the structs,
fix the python data such that the autogen correctly writes out the desired printing
logic.

charles-lunarg committed:
apidump: Consolidate api_dump.cpp template

The intent of these changes is to reduce the copy-pasting between the functions
which have specific code that needs to be executed.

This includes:
 * Combining void and non-void returning functions
 * Moving the special bits of the following functions into the common generator:
    - vkDestroyInstance
	- vkDestroyDevice
	- vkQueuePresentKHR
	- vkGetPhysicalDeviceToolPropertiesEXT
 * Correctly identifying whether a function is instance, device, or global
   - This prevents generation of global functions which all have hardcoded implementations

This fixes an issue where vkGetPhysicalDeviceToolPropertiesEXT didn't unlock
the mutex at the end of the function. Issues like that are why consolidation
was done.

TonyBarbour committed:
build: Update to 1.3.236 headers

christophe-lunarg committed:
vkconfig: fix macOS hang (maybe)

Change-Id: I5d863842c43ccb54708561818c93c39918aafcbd

christophe-lunarg committed:
vkconfig: Add Imagination Technologies best practices

Change-Id: Ia2a9b73012706cfec2dc1310e0f257d6337e18b8

juan-lunarg committed:
vkvia: Fix not finding Ubuntu package on Ubuntu based distros

juan-lunarg committed:
cmake: Fix Wayland CMake warnings

Use pkg-config to find linux system libraries

TonyBarbour committed:
build: Update to 1.3.238 headers

jeremyg-lunarg committed:
layers: Update to use latest vk_layer_logging.h

The api provided by this file has changed.

juan-lunarg committed:
cmake: Raise minimum to 3.17.2

The vulkan validation layers update their CMake min to 3.17.2

Raise min accordingly

juan-lunarg committed:
cmake: Cleanup CMake build

- Update known_good.json to working dependencies
- Delete cmake/FindVulkanHeaders.cmake
- Use Vulkan::Headers
- Use CMake functionality provided by VulkanHeaders
- Remove usage of deprecated vk_sdk_platform.h

juan-lunarg committed:
github: Rename master to main

TonyBarbour committed:
build: Update to 1.3.239 headers

bionade24 committed:
docs: Fix type in vkconfig/README.md

juan-lunarg committed:
cmake: Remove custom install code

charles-lunarg committed:
apidump: Clean up stream handling & update copyright

The logic for stream() can be simplified by putting it into a single
std::ostream object, as well as the ApiDumpSettings can be directly
placed inside the ApiDumpInstance class rather than having it be a
pointer. Putting the settings object nessecitated making the static
object live inside the static current function, so that its
construction happened in the first call to current(), rather than
at an arbitrary time during initialization.

charles-lunarg committed:
apidump: Cleanup threadID() logic

The code used a fixed sized array which obfuscated the logic driving
the function. By using an unordered_map, it not only clarifies the
behavior by using a map but also may help with contention when there
are many threads.

charles-lunarg committed:
apidump: Use boolean or instead of bitwise

While the code ultimately did 'the right thing' using a bitwise operator
looks weird in the context of boolean expressions.

charles-lunarg committed:
apidump: Expand settings file boolean fields

Because there are many ways to express 'true', this commit allows users to
specify a boolean setting value in a variety of ways.

Supported ways are `true`, `on`, and `1`. All are case insensitive.

charles-lunarg committed:
apidump: Remove spaces in json output

There were spaces being added accidentally. This commit fixes that.

charles-lunarg committed:
apidump: Print pNext chain with no_addr set

Turns out there was a small logic error which caused all pNext chains to not
be printed in the text output mode if no_addr was enabled.

charles-lunarg committed:
apidump: Use correct heapBudgetCount in pNext chain

VkPhysicalDeviceMemoryBudgetPropertiesEXT has a fixed size array of
heapBudgets and heapUsages but with a dynamic 'actual' length, based
on the memoryHeapCount of VkPhysicalDeviceMemoryProperties. This
commit adds the necessary logic to allow getting the 'actual' length
so that printing this structure yields more correct results.

charles-lunarg committed:
apidump: Print correct pNext type

The pNext type is either a void* or a const void*. In many places apidump was
using the incorrect one, either because it was hardcoded, or it was using the
type of the child.

charles-lunarg committed:
apidump: Print pointer system types correctly

Certain system types are pointers but were being printed as if they were
values. This was causesing no_address mode to not sanitize the pointer's
address.

charles-lunarg committed:
apidump: fix queue priorities printing

The priorities member of VkQueueFamilyGlobalPriorityPropertiesKHR is a fixed
sized array that uses the priorityCount member to determine the 'real' size.
Thus, apidump should use the priorities member to determine the real length
when printing.

charles-lunarg committed:
apidump: Support Video Extensions

This commit enables API Dump support of the Vulkan Video extensions. The mechanism
requires generating 3 new headers, one for each output type of API Dump using the
video.xml as the source. Because vk.xml and video.xml use the same schema, most of
the generation code works perfectly. A few things needed tweaking, mainly due to
duplicate definitions from both xml's providing definitions for the same types
(like uint32_t).

This commit also modifies the CMake code that handles API Dump to be a little bit
cleaner by centralizing calls to add_dependencies and removing unnecessary if blocks.

charles-lunarg committed:
apidump: Print JSON correctly when detailed is false

When the detailed option is set to false, the JSON output would add trailing
commas to the returnType regardless of whether there was a following element,
which is invalid JSON.

Also, the manually generated UNUSED member sections were moved to a function
which is cleaner to read.

charles-lunarg committed:
apidump: Store data before printing anything

Inside of the struct/function printing functions, there may be one member
which requires another member's data to produce correct output but may not
be accessible within the struct. This means that there are various 'set'
calls made to store data used in a pNext chain or in a subobject. The
original implementation had each 'set' function right before the data it was
relevant to. Unfortunately, there were situations where the data was needed
before the set was called.

The solution is to move the 'set' to the top of the printing functions before
any 'get' has a chance to make use of it.

charles-lunarg committed:
apidump: Fix type_size being ignored

type_size is usually zero but if it wasn't and api dump was outputting with tabs for
fill characters, the padding wasn't being applied. Adding a `<< ""` forces the correct
mode.

When not using tabs for fill, we can use `std::left << std::setw(type_size)` to remove
a strlen call.

charles-lunarg committed:
apidump: Use "address" for appropriate system types

Various system types come from external API's and represent data which
is address-like. This means from run to run the values are changing,
which makes diffing Api Dump output difficult due to spurious changes.
This commit treats all system types like a pointer in that "address"
will be printed if show_address mode is false.

charles-lunarg committed:
apidump: Don't lock for functions that may block

Originally API dump had no issues with blocking API calls, because it would only take the
lock for printing after the function calls returned. But since the change to print the
function name before calling down the chain (in case the function crashes) required printing,
that meant the lock was taken before calling down the chain. If that function blocks then this
stalls EVERY other API call from progressing, which may cause a dead lock.

The solution is to manually specify the functions which may block and have those functions
revert to the original behavior (not print before calling down). This prevents deadlocks by
not holding the lock during blocking Vulkan function calls.

charles-lunarg committed:
Remove usage of VK_LAYER_EXPORT

This PR fixes API Dump, Monitor, Screenshot, and the vk_layer_settings utility to no longer
use the VK_LAYER_EXPORT macro which was defined in vk_layer.h. The macro was recently removed
thus all the projects using it must be updated.

In addition to defining the macro locally to each file, this commit removes function exporting
for functions which do not need it. For Example, it was being added to each and every function
in API Dump when only ~7 functions in the entire Vulkan interface need to be exported.
Additionally, it was entirely removed from the vk_layer_settings utility due to the utility
being a static library thus having no effect.

TonyBarbour committed:
build: Update to 1.3.241 headers

TonyBarbour committed:
build: Remove xxhash dependency from layer factory

TonyBarbour committed:
build: Update to 1.3.243 headers

christophe-lunarg committed:
vkconfig: fix validation layer with generic layer settings

Change-Id: I650d507463695f71ad11096392aab7b3c9aeacb6

christophe-lunarg committed:
vkconfig: updated version

Change-Id: I93d129acc53427a02649481582d4397921b233b0

ncesario-lunarg committed:
build: Use apksigner for Android

ncesario-lunarg committed:
build: Update VVL paths

Updates include paths based on VVL refactor.

Temporary workaround until the VkLayerUtils library is available.

christophe-lunarg committed:
vkconfig: fix copyright

Change-Id: I71fba6a6f14728a5378f9559d6277c8ad5adf8dc

charles-lunarg committed:
apidump: Generate is_struct/is_union logic

Move the logic for dertermining if something is a struct or union to be a
hardcoded value instead of having to perform a runtime lookup. This is a
possibility since whether a type is a struct or union is apart of the Vulkan
API.

This fixes an issue introduced in the PR to support the video extensions where
there is_struct & is_union functions only included the video API types, not
the entire API.

christophe-lunarg committed:
vkconfig: fix link after doc reorganization

Change-Id: I4fd2bde8d09a470f541659b1a57f143384c200ee

christophe-lunarg committed:
vkconfig: fix link after doc reorganization

Change-Id: Ic9d8760fff8e130d9f8db5ef0f46bc6b7e0b4e8b

TonyBarbour committed:
build: Bump VVL known good for next SDK

christophe-lunarg committed:
vkconfig: add display of deviceUUID and driverUUID

christophe-lunarg committed:
vkconfig: add built-in variation to list physical devices

Change-Id: If5015161ecccc491b77805dde57cbbd9ac8a659d

christophe-lunarg committed:
vkconfig: fix Reload Default when unselected config #1811

Change-Id: I3ba6385ec06392abaa928ef6c3e5d4c40b6ed108

christophe-lunarg committed:
vkconfig: fix force physical device

Change-Id: Ic3db8f511a920d747477509010a2ff7a7ca13c49

christophe-lunarg committed:
vkconfig: add default physical device selection config

Change-Id: I2220f5ef0ba97ecaeb951fcee7484f2ad2d2c374

jmacnak committed:
Make update_object_name_map() update on name re-assignment

The emplace() function will not update the value in the map if it
is already present.

davidlunarg committed:
vkconfig doc: Change output of "Setting Type" & "Setting Default Value"

Changed code so these values are displayed on separate lines

charles-lunarg committed:
android ci: Use specific validation layer commit

During various fixes to this repo, the android CI version was moved to main, which
fixed the issues that we were seeing at the time. Now that the dust has settled, its
time to move it back to a stable version. The version was chosen corresponding to
about where the validation was when this repo was set to main.

pingpongcat committed:
Update version.h

Fix for multiple compilation errors on Fedora 38 g++ (GCC) 13.0.1 due to the lack of cstdint include

TonyBarbour committed:
build: Update to 1.3.249 headers

TonyBarbour committed:
build: Use python3 specifically in update scripts

MarkY-LunarG committed:
Fix JSON output on Mac M1

The MacOS on the M1 produced a product version that seems to have a tab
in it which messes up VkVia.  Attempt to fix this.

TonyBarbour committed:
build: Update to 1.3.250 headers

pierremoreau committed:
build: Address typos in BUILD.md

pierremoreau committed:
vkconfig: Address typos in CHANGELOG.md

christophe-lunarg committed:
vkconfig: fix validation layer settings

Change-Id: I5f6434d125b56eb1586471005df9d2cea990aec3

christophe-lunarg committed:
vkconfig: fix QFSFileEngine::open: No file name specified messages

Change-Id: I4ab20ab3094ec15b9704f455af229822669ca1a3

christophe-lunarg committed:
vkconfig: fix profile name invalid init

Change-Id: I25493632e09a00aff42baf4c8f6ea33f9a497fea

christophe-lunarg committed:
cmake: move jsoncpp to knowngood to simply build steps

Change-Id: I49729560d3929a88539b8051c1a60719e1ea76ec

christophe-lunarg committed:
cmake: add Vulkan-Utility-Libraries dependence

Change-Id: Ifacbef603d46c0fed7e33f026fec65f7ccb30ea1

06393993 committed:
apidump: Use logcat instead of stdout for Android

Make the apidump output to go to logcat instead of stdout on Android.

* Implement the AndroidLogcatBuf class that derives from std::streambuf
  which writes to the Android logcat on a per-line basis.
* Modify ApiDumpSettings to hold an instance of AndroidLogcatBuf, and
  make the output_stream to use that streambuf instead of the iostream
  one at initialization.

06393993 committed:
screenshot: Fix Android build error

... unused variable. Without this commit, the build will fail with the
following message:

```
make: *** [%LOCAL_APPDATA%/Android/Sdk/ndk/25.2.9519653/build/../build/core/build-binary.mk:424: obj/local/x86/objs/VkLayer_screenshot/__/__/layersvt/screenshot.o] Error 1
[x86_64] Compile++      : layer_utils <= vk_layer_config.cpp
[x86_64] Compile++      : VkLayer_screenshot <= screenshot.cpp
jni/../../layersvt/screenshot.cpp:518:10: error: variable 'pass' set but not used [-Werror,-Wunused-but-set-variable]
    bool pass;
         ^
1 error generated.
```

charles-lunarg committed:
monitor: Fix windows crash when using renderdoc

The monitor layer uses GetWindowText to retrieve the title bar text.
This function is effectively SendMessage(WM_GETTEXT), and so has all
the same threading behavior of SendMessage. That is fine if the thread
calling into monitor layer is able to handle it. But in certain
circumstances, calling GetWindowText causes a deadlock, such as when
the monitor layer is enabled during the replay of a renderdoc capture.

The fix is to move the call of GetWindowText to vkQueuePresentKHR, since
that does get called from a thread that can handle GetWindowText.

A more robust method would be to not use GetWindowText but instead use
Windows API calls that have better threading characteristics.

charles-lunarg committed:
apidump: Use correct aliased types

VkAccelerationStructureTypeNV and VkBuildAccelerationStructureFlagsNV are
aliased types which are explicitly used by the VK_NV_ray_tracing extension.
Because apidump didn't know how to keep track of aliases and then fixup
uses of aliased types into the real type, those two types had hand written
typedefs. This commit is able to remove the hardcoding as well are removes
both redundant code (self.trackedTypes) and dead code
(str_VkStructureTypeToEnum).

charles-lunarg committed:
via: Dont set the MSVC runtime library

The update to CMake 3.15+ changes how the MSVC runtime library
(as in the CRT) should be set. Instead of modifying C_FLAGS, there is a
property per target that must be set. Since nothing in this repo stands
to benefit from statically linking the CRT, no additions are needed.
Only the removal of the useless CMake foreach loop is necessary.

charles-lunarg committed:
via: Remove CMakeLists-SDK.txt

This was used for old SDK builds but is no longer in use. It is out of
date and so should be removed.

TonyBarbour committed:
build: Update to 1.3.257 headers

Also remove VK_NV_cooperative_matrix and VK_QNX_external_memory_screen_buffer
from codegen

Also update Qt installer version

spencer-lunarg committed:
layers: Cleanup unused includes and bool expr

TonyBarbour committed:
build: Update to 1.3.260 headers

Skip updating Android due to VVL Android cmake build

TonyBarbour committed:
build: Update to 1.3.261 headers

Had to use a  VVL commit that reverted code that prevented
building VVL in this repo

charles-lunarg committed:
repo: Remove Vulkan-Layer-Factory

The Vulkan-Layer-Factory project is being moved into its own repo for
archival purposes. This commit removes it from this repo as well as
documents where the archival repo is.

charles-lunarg committed:
docs: Update Readme

Remove notification of default branch change & CI status badges from the readme.

charles-lunarg committed:
api_dump: Prevent crash in VkDescriptorDataEXT union

The VK_EXT_descriptor_buffer extension adds a union with pointer elements.
Because API Dumps union logic is naive, it prints the value of every element
in the union regardless of whichever the 'active' element is, as it is not
always possible to easily get the active element. VkDescriptorDataEXT
contains pointers which are dereferenced resulting in a crash. Fixing it is
straightforward because the parent structure VkDescriptorGetInfoEXT contains
the type to use, so it is just a matter using the PARAMETER_STATE and
VALIDITY_CHECKS functionality to set & get the type when printing out the
contents of the union.

juan-lunarg committed:
Remove unused CMake build

juan-lunarg committed:
Ignore CMakeUserPresets.json

juan-lunarg committed:
Minor CMake cleanup

Remove USE_CCACHE
Remove CMAKE_OSX_DEPLOYMENT_TARGET
Remove FALLBACK_CONFIG_DIRS / FALLBACK_DATA_DIRS

Properly set enable_testing()

juan-lunarg committed:
ci: Cleanup GitHub Actions CI

Fixup performance issues
Fixup CI warnings

juan-lunarg committed:
Delete CMakeUserPresets.json

charles-lunarg committed:
apidump: Query device functions during GetInstanceProcAddr

vkGetInstanceProcAddr is able to query device functions, which was not being
done by api_dump. This caused debug utils functions to not be intercepted due
to changes in how the loader queries those functions.

charles-lunarg committed:
Enable VK_NV_cooperative_matrix

The extension was disabled due to issues with the transition from an NV
ext to a KHR ext. This required a fix to the autogeneration, which was
previously committed.

charles-lunarg committed:
Enable VK_QNX_external_memory_screen_buffer

This extension was disabled due to not working with API dump, to which this
commit fixes. The souce of the error was that vulkan safety critical modified
vk.xml to change the supported="vulkan" to allow supported="vulkan,vulkansc".
ApiDump's generator was only looking for vulkan, causing any extension that
worked with both API's to be missed, causing the extension guard to not be
added where required.

This commit also adds the SCI platform define to the hardcoded list.

charles-lunarg committed:
build: Use Vulkan-Loader on windows

Previously, the vulkan-1.lib file was checked into the tree, so that it wasn't
a required build step for windows builds. This means that to use functions in
new versions of vulkan, the vulkan-1.lib file would have to be updated, which
is annoying.

This commit makes the necessary changes to allow linking to the vulkan-1.lib
built by Vulkan-Loader directly, as well as adding Vulkan-Loader as a
dependency for windows builds. It was previously only built for linux & macOS.

With the use of the Vulkan-Loader, the FindVulkan.cmake file could be removed,
as well as the checked in vulkan-1.lib files, one each for x86 and x64.

juan-lunarg committed:
cmake: Remove cmake directory

We don't use these scripts

juan-lunarg committed:
cmake: Cleanup Python Generation Code

find_package(PythonInterp) is deprecated.
Use find_package(Python3).

Move codegen code to layersvt/CMakeLists.txt where it is actually
used.

Create a VULKAN_REGISTRY variable to reduce code duplication.

Remove pointless cmake_minimum_required

juan-lunarg committed:
cmake: Remove excess cmake_minimum_required / project usage

juan-lunarg committed:
Remove external/googletest

juan-lunarg committed:
update update_deps.py

juan-lunarg committed:
Add googletest to known_good.json

juan-lunarg committed:
Utilize known_good.json for jsoncpp / valijson

Remove external/submodule directories

closes #1761

juan-lunarg committed:
cmake: Fix -Wno-maybe-uninitialized

Not needed now that we grab jsoncpp via known_good.json

juan-lunarg committed:
Delete determine_vs_version.py

closes #558

juan-lunarg committed:
cmake: Add UPDATE_DEPS support

juan-lunarg committed:
Get Android CMake build working

juan-lunarg committed:
Remove build-android

juan-lunarg committed:
CMake ensure important compiler flags/definitions are set

Ensure CMAKE_POSITION_INDEPENDENT_CODE is set uniformly
Ensure VISIBILITY is set uniformly
Ensure beta extensions and android definitions are set uniformly

Needed to ensure the Android builds are on parity with the
removed ndk-build

juan-lunarg committed:
docs: Remove deprecated Android build instructions

charles-lunarg committed:
layers: Use Vulkan-Utility-Libraries for dispatch table

Use the dispatch tables provided by the Vulkan-Utility-Libraries'
UtilityHeaders target rather than the tables from Validation Layers.
This is the first commit in removing ValidationLayers as a dependency.

juan-lunarg committed:
Fix known_good.json

christophe-lunarg committed:
layer: Use layer settings library

christophe-lunarg committed:
cmake: Enable MSVC parallel build

Change-Id: I557e7fdeff4fa27b39d5a50829eba015fb8cae4a

christophe-lunarg committed:
vkconfig: Use layer settings library

christophe-lunarg committed:
layers: Use layer settings library in layers code

mgorchak-blackberry committed:
Add QNX types.

richard-lunarg committed:
Add case for iOS builds with cmake

christophe-lunarg committed:
cmake: Add layer files to solution

Change-Id: I32b991f72052a58bc35d514d49c64ea6c82594d3

christophe-lunarg committed:
api-dump: Use layer settings library

christophe-lunarg committed:
vkconfig: Fix warnings

Change-Id: Ifbe8f1af03d737e496caf54e78d26a995b8be07c

juan-lunarg committed:
build: Remove VVL as a dependency

christophe-lunarg committed:
cmake: Update valijson version

Change-Id: I2247dcfe860055f1269ae14b8b69b826811c0ed8

charles-lunarg committed:
apidump: Fix regression in the NoAddr setting

api_dump.h stores the no-address setting in a variable called show_address.
This requires inverting the setting to match what the rest of the api_dump
code expects it to mean. This fixes a regression introduced by the transition
to using the Layer Settings library from Vulkan-Utility-Libraries.

charles-lunarg committed:
build: Use Vulkan-Loader find_package support

Simplifies the CMake code since Vulkan-Loader now exports itself in an
easy to consume format. No need to create our own target, just use the
provided one.

pixelcluster committed:
apidump: Return NULL in GDPA for disabled extensions

vkGetDeviceProcAddr specifies that only function pointers of enabled
extensions should be exposed. If an extension is not enabled,
vkGetDeviceProcAddr should return NULL. Some users like winevulkan rely
on this behaviour and break when used with the current API dump layer.

Assuming that drivers and other layers act accordingly and return NULL
for non-enabled entrypoints, we can mirror this by simply returning
NULL when the corresponding dispatch table entry is NULL.

pixelcluster committed:
apidump: Return NULL in GIPA for disabled extensions

vkGetInstanceProcAddr specifies that only function pointers of available
extensions should be exposed. If an extension is not available,
vkGetDeviceProcAddr should return NULL.

Similar to how it is done for GDPA, we can check if the function pointer
is non-NULL in the instance dispatch table.

charles-lunarg committed:
layers: Update to new names in vk_dispatch_table.h

The functions and structs in vk_dispatch_table.h have been updated to use a
'vku' prefix instead of 'vul'. Thus, this repo needs to update to it so the
next header update goes smoothly.

charles-lunarg committed:
apidump: Check for Graphics Pipeline Library properly.

The members of VkPipelineRenderingCreateInfo should be ignored (except for viewMask)
when either VK_GRAPHICS_PIPELINE_LIBRARY_PRE_RASTERIZATION_SHADERS_BIT_EXT or
VK_GRAPHICS_PIPELINE_LIBRARY_FRAGMENT_SHADER_BIT_EXT is set in the flags of
VkGraphicsPipelineLibraryCreateInfoEXT.

Doing this requires looking for the VkGraphicsPipelineLibraryCreateInfoEXT struct
in the pNext chain of VkGraphicsPipelineCreateInfo.

This commit also cleans up a few things:
 * Use enum name instead of value in pNext_trampolines for loader structs
 * Eliminate the name & type fields in the PARAMETER_STATE since they were unused.
 * Remove whitespace

charles-lunarg committed:
deps: Update to latest Vulkan-Utility-Libraries

Because the layer settings library changes its prefix to vku, all uses of it in
this repo must be updated.

charles-lunarg committed:
apidump: Fix regression in Environment Variable naming

The use of the layer settings library changed the established names for
the settings set through environment variables. This commit addresses
this by using the compatibility namespace feature to ensure existing
users aren't affected.

Also, because the settings are initialized during initLayerSettings,
the Thread and Frame setting was unset during the printing of
vkCreateInstance's head. Moving the initialization to before printing
resolves this regression.

TonyBarbour committed:
build: Update to 1.3.265 headers

juan-lunarg committed:
tests: Update to googletest v1.14.0

TonyBarbour committed:
build: Update to 1.3.268 headers

zangold-nv committed:
Fix some issues from 1.3.268 merge

zangold-nv committed:
allow json_gen layer to be built independently of api_dump

zangold-nv committed:
Merge tag 'v1.3.268' into zja-sdk-fixes-2

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/VulkanTools-ForSC?pullRequest=15) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/VulkanTools-ForSC?pullRequest=15) before we can accept your contribution.<br/>**0** out of **4** committers have signed the CLA.<br/><br/>:x: pixelcluster<br/>:x: juan-lunarg<br/>:x: TonyBarbour<br/>:x: mgorchak-blackberry<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/VulkanTools-ForSC?pullRequest=15) it.</sub>

zangold-nv committed:
Only look for OpenSSL when building vkjson layer

zangold-nv committed:
several fixes to json_gen layer tests

- fix some incorrect helptext in the json_gen layer tests

- fix a bug where the test executable was being called without any
  arguments (because shell=True needs args to be a single string,
  whereas shell=False accepts a list of strings corresponding to
  individual arguments)

- exit the test executable with nonzero exit code when any of the vulkan
  commands fail

- Fail the test when the executable returns with nonzero exit codes

- prevent false test success when the test doesn't create any files

- fix an issue where the compute shader was referencing an unbound
  uniform. Update the expected/golden JSON output, and the expected
  pipeline UUID.

- Update the expected pipeline UUID for the graphics pipeline test,
  since it didn't match the pipeline UUID in the golden JSON.

- fix hardcoded windows path separator ('\')

- fix a memory leak (compute pipeline layout)

dgkoch commented on jsonGenLayerTests/main.cpp:
```diff
@@ -456,24 +411,21 @@ bool checkPipelinePropertiesEXT()
-        std::cerr << "Error: vkGetPipelinePropertiesEXT failed to produce correct UUID" << std::endl;
-        return false;
-    }
-
```
Curious why this was deleted?

We no longer do anything with the retrieved UUIDs.

dgkoch replied:
(oh - I see your comment in the commit about the being verified by the wrapping script).

dgkoch replied:
Actually - we need to restore this check - this is verifying that the VK_EXT_pipeline_properties layer is working correctly.

dgkoch reviewed (approved):
LGTM
